### PR TITLE
feat(style): add more complete parsing of vector tiles style

### PR DESCRIPTION
**Pull request by zarov**
About half of symbol and text style has been added for a Vector Tiles
style. See the specification here
https://docs.mapbox.com/mapbox-gl-js/style-spec/

Usage of this style will come in future PRs.

### CODE REVIEW

File: src/Core/Style.js [hunk 1]
```diff
@@ -131,23 +138,79 @@ class Style {
...
+            this.text.halo = {};
+            this.text.halo.color = readVectorProperty(layer.paint && layer.paint['text-halo-color'] || '#000000', zoom);
+            this.text.halo.width = readVectorProperty(layer.paint && layer.paint['text-halo-width'] || 0, zoom);
+            this.text.halo.blur = readVectorProperty(layer.paint && layer.paint['text-halo-blur'] || 0, zoom);
+
+            // additional icon
+            const iconSrc = readVectorProperty(layer.layout && layer.layout['icon-image'], zoom);
+            if (iconSrc) {
+                let size = readVectorProperty(layer.layout && layer.layout['icon-size'], zoom);
+                if (size == undefined) { size = 1; }
```
Comment (gchoqueux) [inline src/Core/Style.js:195]: `const size = readVectorProperty(layer.layout && layer.layout['icon-size'] || 1, zoom);`

File: src/Core/Style.js [hunk 2]
```diff
@@ -131,23 +138,79 @@ class Style {
...
+
+                const icon = Cache.get(`${iconSrc}-${size}`);
```
Comment (gchoqueux) [inline src/Core/Style.js]: set directly `this.icon = Cache.get(`${iconSrc}-${size}`);`

File: src/Core/Style.js [hunk 3]
```diff
@@ -131,23 +138,79 @@ class Style {
...
+
+                if (icon) {
+                    this.icon = icon;
+                } else {
+                    const sprite = sprites[iconSrc];
+                    canvas.width = sprite.width * size;
+                    canvas.height = sprite.height * size;
+                    canvas.getContext('2d').drawImage(sprites.img, sprite.x, sprite.y, sprite.width, sprite.height, 0, 0, sprite.width * size, sprite.height * size);
```
Comment (gchoqueux) [inline src/Core/Style.js]: could you factorize this part with sprite?

File: src/Core/Style.js [hunk 4]
```diff
@@ -131,23 +138,79 @@ class Style {
...
+                    this.icon = {};
+                    this.icon.dom = document.createElement('img');
+                    this.icon.dom.src = canvas.toDataURL('image/png');
+
+                    this.icon.anchor = readVectorProperty(layer.layout && layer.layout['icon-anchor'] || 'center', zoom);
+                    this.icon.halfWidth = canvas.width / 2;
+                    this.icon.halfHeight = canvas.height / 2;
+                    // this.icon.padding
```
Comment (gchoqueux) [inline src/Core/Style.js]: `padding`?

File: src/Core/Style.js (hunk 1 above)
Comment (zarov) [reply src/Core/Style.js:195]: I can't, because it can be equal to `0`.

File: src/Core/Style.js (hunk 2 above)
Comment (zarov) [reply src/Core/Style.js]: Done

File: src/Core/Style.js (hunk 3 above)
Comment (zarov) [reply src/Core/Style.js]: Yes

File: src/Core/Style.js (hunk 4 above)
Comment (zarov) [reply src/Core/Style.js]: Woops

**Review by gchoqueux — APPROVED** [review]

**Pull request merged**